receiver/scraper/error/doc.go

```diff
@@ -2,4 +2,4 @@
 // SPDX-License-Identifier: Apache-2.0
 
 // Package scrapererror provides custom error types for scrapers.
-package scrapererror // import "go.opentelemetry.io/collector/receiver/scrapererror"
+package error // import "go.opentelemetry.io/collector/receiver/scraper/error"
```

Comment:
Can we keep this as scrapererror? (Here and in other files?) And also name the folder `scrapererror`?

`error` is (I think) already a common package name so to avoid collisions and to help existing users to migrate more easily, I would rather keep the `scrapererror` name